c2rust-analyze/src/context.rs

```diff
@@ -182,6 +193,27 @@ impl<'tcx> GlobalAnalysisCtxt<'tcx> {
             self.assign_pointer_to_field(field);
         }
     }
+
+    pub fn fn_failed(&mut self, did: DefId) -> bool {
+        self.fns_failed.contains_key(&did)
+    }
+
+    pub fn mark_fn_failed(&mut self, did: DefId, detail: PanicDetail) {
+        if self.fns_failed.contains_key(&did) {
+            return;
+        }
+
+        self.fns_failed.insert(did, detail);
```

Comment:
Do we always want to skip a 2nd error?  Could the 2nd error be a panicking error, and the 1st one just be a propagated one?  In that case, the panicking error would be more useful to keep.

Reply:
We could also just store all of the `PanicDetails`.

Reply:
Since we visit functions in callgraph order (callee first, then caller), in nearly all cases where there's both a propagated error and an error in the caller itself, we will set the propagated error first and then never analyze the caller (so we won't see the non-propagated error at all).  The only exception is for mutually recursive functions.

We'll soon be getting rid of this style of error propagation through the callgraph, and I think it's fine to leave this as-is for now.